src/Templates/AdminPanel/Bit.AdminPanel/Web/Components/AppComponentBase.cs

```diff
@@ -1,8 +1,28 @@
-﻿namespace AdminPanel.App.Components;
+﻿using Microsoft.Extensions.Localization;
+
+namespace AdminPanel.App.Components;
 
 public partial class AppComponentBase : ComponentBase
 {
-    [AutoInject] private IExceptionHandler exceptionHandler = default!;
+    [AutoInject] protected IExceptionHandler ExceptionHandler { get; set; } = default!;
+
+    [AutoInject] protected IStateService StateService = default!;
+
+    [AutoInject] protected AppAuthenticationStateProvider AuthenticationStateProvider = default!;
+
+    [AutoInject] protected HttpClient HttpClient = default!;
+
+    [AutoInject] protected IAuthTokenProvider AuthTokenProvider = default!;
+
+#if BlazorServer || BlazorHybrid
+    [AutoInject] protected IConfiguration Configuration = default!;
+#endif
+
+    [AutoInject] protected NavigationManager NavigationManager = default!;
+
+    [AutoInject] protected IAuthenticationService AuthenticationService = default!;
+
+    [AutoInject] protected IJSRuntime JavaScriptRuntime = default!;
```

Comment:
Rename it to `JSRuntime` and do the same in `TodoTemplate` and remove usages of IJSRuntime in inherited components / pages